### ISSUE TITLE
babel preset es2015 & es2015-riot

### ISSUE THREAD

**Issue by fractalf** [riot/compiler#75]
I've been fighting for some hours now to make riot use es6 js..

````
<script type="babel">
  console.log(this)
</script>
````

*this* is *undefined* which is kind of covered here https://github.com/riot/compiler/issues/41

basically using es2015-riot instead of the normal es2015 preset, and that actually works.
..btw, they are very simular except the riot version has a bunch of extra `{'loose': true}` options, weird.

However, I also have some normal .js files I would just like to use the normal es2015 preset.
So I removed the preset from .babelrc and moved them into webpack.config.js like this
````
...
{
    test: /\.js$/,
    loader: 'babel',
    query: { presets: ['es2015'] }
},
{
    test: /\.tag$/,
    loader: 'riotjs',
    query: { presets: ['es2015-riot'] }
},
...
````
...but that just don't work, now the .tag script dont get transpiled at all.

Anyone had simular problems?

**Comment by GianlucaGuarini** [comment]
Hi @fractalf you can read a bit more about how to configure riot with babel 6 on our site http://riotjs.com/guide/compiler/#babel-6. I have created a repo to show how to configure riot in an isomorphic app totally build with es6 syntax https://github.com/GianlucaGuarini/riot-app-example. I hope you will be able to configure riot just enabling the following option in your babel config file:
```json
{
  "plugins": [
    ["transform-es2015-modules-commonjs", { "allowTopLevelThis": true }]
  ]
}
```

**Comment by fractalf** [comment]
Hi @GianlucaGuarini  and thanks for your reply. I did read that doc more than once and I did actually have that plugin set, but took it away because of something I read in https://github.com/riot/compiler/issues/41 posted above:

> You need to remove "transform-es2015-modules-commonjs" from your plugins
> -- GianlucaGuarini

I'm guessing now that this might have something to do with webpack, because this *.babelrc* works fine:
````
{
    "presets": [
        "es2015-riot",
        "stage-0",
        "stage-2",
        "stage-3",
    ],
    "plugins": [
        "transform-runtime",
    ]
}
````
But like I said, I'm not sure how well this setup would affect normal es6 .js files transpiling, so that's why I tried to move this setting from *.babelrc* to *webpack.config.js* with two seperate presets.

I'm hoping someone else worked with webpack and riotjs and have been in the same situation..

**Comment by fractalf** [comment]
Hm... thanks, but I can't make sense of that

Those plugins you mention in .babelrc are exactly the same as in the "es2015-riot" preset? Altso I'm trying *not to* use presets/plugins in *.babelrc* so I can use different presets/plugins for transpiling .js and .tag.

About the webpack config suggestion..
https://github.com/babel/babel-loader:
> cacheDirectory: Default false. When set, the given directory will be used to cache the results of the loader. Future webpack builds will attempt to read from the cache to avoid needing to run the potentially expensive Babel recompilation process on each run

I don't understand what this has got to do with anything ^

`loader: 'tag'`  ..so there's actually two loaders... I've been using this [riotjs-loader](https://github.com/esnunes/riotjs-loader). Tried it, but not much difference.

I appresiate the help, but I want to understand whats going on here, not just try "a lot of different stuff" :)

I mean, why don't this actually work:

````
<script>
    let foobar = 0;
</script>
````

````
..
{
    test: /\.js$/,
    exclude: /node_modules/,
    loader: 'babel',
    query: {
        presets: ['es2015']
    }
},
{
    test: /\.tag$/,
    exclude: /node_modules/,
    loader: 'tag',
    query: {
        type: 'babel',
        presets: ['es2015-riot']
    }
},
..
````

**Comment by GianlucaGuarini** [comment]
@fractalf I don't know what's the problem in your case but I made a test and everything works just fine https://github.com/GianlucaGuarini/riot-webpack-babel-template - Maybe you could submit the issue to the babel/webpack community

**Comment by fractalf** [comment]
@GianlucaGuarini yes your example is the same that I described and all apparently "works", but how can I be sure?

My "problem" with this is
- how can I be sure this doesn't mess up normal es6 .js transpilation?

Like I described the *es2015* and *es2015-riot* are almost identical, but the first one don't work with .tag files. So how can I be sure the second one don't somehow mess up when transpiling .js files.

But you are right, this doesn't look like a riotjs problem så I'll try refrase the problem a bit with more example code and post it in the webpack repo.

Thanks for your help m8 :)
